pilot/pkg/model/push_context.go

```diff
@@ -613,6 +617,31 @@ func (ps *PushContext) initVirtualServices(env *Environment) error {
 	return nil
 }
 
+// Caches list of Sidecar resources
+func (ps *PushContext) initSidecarScopes(env *Environment) error {
+	sidecarConfigs, err := env.List(Sidecar.Type, NamespaceAll)
+	if err != nil {
+		return err
+	}
+
+	sortConfigByCreationTime(sidecarConfigs)
+
+	for _, sidecarConfig := range sidecarConfigs {
+		// TODO: add entries with workloadSelectors first before adding namespace-wide entries
+		ps.sidecarsByNamespace[sidecarConfig.Namespace] = append(ps.sidecarsByNamespace[sidecarConfig.Namespace], ConvertToSidecarScope(ps, &sidecarConfig))
```

Comment:
SidecarScope is that we already want.  But We need scalability test to seem how this influence pilot: configuration push latency?

Reply:
I think the delay will be in reading and parsing these additional CRDs and forming all the maps. But the config push latency will be lower because with a sidecar scope, we can limit the number of listeners/clusters/routes that we push to each envoy.